### ISSUE TITLE
Mysterious overflow error in LiveAuctionBidViewController

### ISSUE THREAD

**Issue by sarahscott** [artsy/eigen#1405]
Not easily reproducible, and inhibits builds but doesn't seem to be an issue at runtime if the compiler allows me to build.

![screen shot 2016-04-12 at 12 01 26 pm](https://cloud.githubusercontent.com/assets/2712962/14468506/e52e54ca-00ad-11e6-8e2b-f236e1feb6a4.png)

**Comment by alloy** [comment]
Because we still support 32-bit devices, e.g. iPad 2, this should really be `Int64`.

**Comment by ashfurrow** [comment]
Interesting – I imagine this is a bug worth reporting to Apple (I'll file a radar) because the problem appears to be pattern-matching an `Int` against a range that the compiler _should_ infer as `Range<Int64>`. So either that inference is wrong or something is up with cases with `Range<Int>` and `Range<Int64>` in the same `switch`. /cc @orta

**Comment by sarahscott** [comment]
FWIW it's not happening for me anymore  ¯\_(ツ)_/¯

**Comment by orta** [comment]
I'm also chill on reducing the number, ruby supports a better `switch` so this was a bit lossy in porting

**Comment by alloy** [comment]
> the problem appears to be pattern-matching an `Int` against a range that the compiler should infer as `Range<Int64>`.

@ashfurrow While that might be true, the code as it is would suggest to me that `bid` _may_ actually be that large, in which case it should explicitly be a `Int64` as well.

However, @orta’s last comment suggests that it might not ever be that high?

**Comment by alloy** [comment]
> I imagine this is a bug worth reporting to Apple (I'll file a radar)

Nonetheless, good call on that :+1: Please include the open-radar link here.

**Comment by ashfurrow** [comment]
Using `Int64` explicitly produces a more helpful error:

```swift
case Range<Int64>(0...100_000_000_000_00): // Expression pattern of type Range<Int64> cannot match values types of 'Int'
```

But again, only when compiled on 32-bit architectures. I'm unsure what the actual bug in the tooling is that I should be reporting...

**Comment by alloy** [comment]
I would expect a number like `100_000_000_000_00` to automatically be `Int64`, maybe it’s not doing that when you compare it against an `Int` on a 32-bit platform (i.e. it’s `Int32`) ?

**Comment by ashfurrow** [comment]
I think what's happening is the `Range`s used in case statements have types inferred from the switch value (in this case, `Int`) instead of the values expression in the case statement. I guess that would be the bug, that I should be able to switch on an `Int32` and have a `Range<Int64>` match it.

**Comment by alloy** [comment]
Yep, that’s what I meant too 👍 Should be easy to verify in a short script.

**Comment by ashfurrow** [comment]
I suppose this is a Swift bug, and I should file a JIRA ticket 🤔🤔🤔

**Comment by alloy** [comment]
@ashfurrow Can you first make a simple report script and paste it here? I _might_ want to take a look at the Swift source for it and don’t want to make any promises about that to them.

**Comment by ashfurrow** [comment]
Sure, that makes sense (I'm still not 100% convinced it's a bug as much as a limitation on Swift 2.2 (it doesn't/can't translate from Int32 to Int64 automatically, same with CGFloat/double I hear).

**Comment by ashfurrow** [comment]
This is related to automatic conversion between numerics, which is [out of scope](https://github.com/apple/swift-evolution/pull/238) for now. I think the errors are descriptive enough that we don't need to file a bug report for this.

**Comment by alloy** [comment]
@ashfurrow As it has not been stated explicitly I just want to be sure this is clear, we won’t be able to do any betas without this being fixed.

**Comment by ashfurrow** [comment]
Right, apologies. I'll send a PR if @orta doesn't beat me to it (he wrote the increment code, I don't imagine it will be difficult to make this change though).

**Comment by orta** [comment]
I'm on it :+1:

**Comment by ashfurrow** [comment]
Shoot, just submitted >.<

**Comment by ashfurrow** [comment]
This is fixed by #1421.
